### ISSUE TITLE
RFE: Collect cluster metadata before triggering a workload

### ISSUE THREAD

**Issue by aakarshg** [cloud-bulldozer/ripsaw#47]
We should look at running stockpile/ other tool to collect cluster metadata before we trigger the workload, instead of having a harness run stockpile/other_tool to collect data, I'd propose that we trigger the playbook/role from within the operator itself.
Possible cons of this would be:
- Longer runs for the operator, as it'll have to finish collecting metadata.
- Requires stockpile/other.tool to be closely integrated with the operator.
- Also require processing/indexing tool to be able to parse the benchmark results with in the operator logic and index the results.

**Comment by jtaleric** [comment]
If we want this in the Operator, I think the RFE #36 will be necessary...

**Comment by aakarshg** [comment]
I believe that #36 is necessary in general. if we're going to collect metadata within operator logic, then we'd not require to read the status, we'll just be doing it when the CR changes. The idea is to move the logic in uperf a level up, so we can do something like this:
```
- hosts: localhost
  gather_facts: no
  tasks:
  - name: Check for current_run
    stat:
      path: /tmp/current_run
    register: state_file

  - debug:
      msg: "State file : {{state_file.stat}}"

  - name: Check for status
    set_fact:
      contents: "{{ lookup('file', '/tmp/current_run') }}"
    when: state_file.stat.exists == true

  - name: Retrieve previous values
    set_fact:
      prev_values: |
        [
        {% for item in contents %}
          "{{ item }}:{{ contents[item] }}",
        {% endfor %}
        ]
      curr_values: |
        [
        {% for item in workload %}
          "{{ item }}:{{ workload[item] }}",
        {% endfor %}
        ]
    when: state_file.stat.exists == true

  - debug:
      msg: "Check : {{curr_values}} vs {{prev_values}}"
    when: state_file.stat.exists == true

- block:
  - name: Include stockpile only if the condition is true
    include: stockpile.yaml

  - hosts: localhost
    gather_facts: no
    tasks:
    - include_role:
        name: "uperf-bench"
      when: uperf is defined and uperf.pair > 0
    - include_role:
        name: "fio-bench"
      when: fio is defined and fio.clients > 0
    - include_role:
        name: sysbench
      when: sysbench is defined and sysbench.enabled is defined and sysbench.enabled
    - include_role:
        name: "couchbase-infra"
      when: couchbase is defined and couchbase.servers.size > 0
 when: ( workload exists and workload|length > 0 ) and ( state_file.stat.exists == false  or (curr_values | difference(prev_values))|length > 0 )
```

**Comment by aakarshg** [comment]
addressed with #240 , closing this.
